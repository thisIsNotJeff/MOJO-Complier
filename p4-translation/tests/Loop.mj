{
    var x := 0;
    
    loop while x < 100 {
        x := x + 1;
        if x == 42 {
            break;
        }
    }
}
