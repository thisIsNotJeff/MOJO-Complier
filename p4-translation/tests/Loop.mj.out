	.text
Loop:
MOVE(
 TEMP Loop.1.x,
 CONST 0),
JUMP(
 NAME L.1),
LABEL L.2,
MOVE(
 TEMP Loop.1.x,
 ADD(
  TEMP Loop.1.x,
  CONST 1)),
BEQ(
 TEMP Loop.1.x,
 CONST 42,
 L.4, L.5),
LABEL L.4,
JUMP(
 NAME L.3),
LABEL L.5,
LABEL L.1,
BLT(
 TEMP Loop.1.x,
 CONST 100,
 L.2, L.3),
LABEL L.3


